app/lib/search/dart_sdk_mem_index.dart

```diff
@@ -61,13 +61,19 @@ Future<SdkMemIndex?> _createDartSdkMemIndex() async {
     return await retry(
       () async {
         final index = SdkMemIndex.dart();
-        final uri = index.baseUri.resolve('index.json');
-        final rs = await http.get(uri);
-        if (rs.statusCode != 200) {
-          throw Exception('Unexpected status code for $uri: ${rs.statusCode}');
-        }
-        final content = DartdocIndex.parseJsonText(rs.body);
+        final content = DartdocIndex.parseJsonText(
+          await searchBackend.fetchSdkIndexContentAsString(
```

Comment:
Using `httpRetryClient` inside `retry`...

Reply:
we are using `httpRetryClient` inside `fetchSdkIndexContentAsString`

Reply:
Yes, I would generally discourage wrapping retries with more retries :D

Since the place we're calling `fetchSdkIndexContentAsString` (here) we are also inside a `retry` callback.